aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -328,6 +330,15 @@ private static InputStream uncheckedInputStream(File file) {
     }
   }
 
+  private void createStagingDirectoryIfNotExists() throws SecurityException {
+    if (!stagingDirectory.exists()) {
```

Comment:
add a LOG saying something like "staging directoy {} not exist, trying to create one"

Reply:
Added the logger.